parteh/PRTLossFluxesMod.F90

```diff
@@ -0,0 +1,659 @@
+      !    or loss of biomass from the death of the plant.
+      ! 3) Since this is maintenance turnover, and not a complete drop of leaves for
+      !    deciduous trees, we just re-translocate nutrients (if necessary) from the
+      !    leaves and roots that leave (no pun intended), into the leaves and roots that
+      !    are still rooted to the plant (pun intended). For deciduous, event-based
+      !    phenology, we will re-translocate to the storage pool.
+      ! 4) There are currently no reaction costs associated with re-translocation
+      ! ---------------------------------------------------------------------------------
+      
+      class(prt_vartypes) :: prt
+      integer,intent(in) :: ipft
+      
+      integer :: i_var
+      integer :: spec_id
+      integer :: organ_id
+      integer :: i_pos
+
+      real(r8) :: turnover
+      real(r8) :: leaf_turnover
+      real(r8) :: fnrt_turnover
+      real(r8) :: sapw_turnover
+      real(r8) :: store_turnover
+      real(r8) :: struct_turnover
+      real(r8) :: repro_turnover
+      real(r8), dimension(num_organ_types) :: base_turnover   ! A temp for the actual turnover removed from pool
+      real(r8) :: retrans    ! A temp for the actual re-translocated mass
+
+      
+      ! -----------------------------------------------------------------------------------
+      ! Calculate the turnover rates (maybe this should be done once in the parameter
+      ! check routine. Perhaps generate a rate in parameters derived?
+      ! -----------------------------------------------------------------------------------
+
+      base_turnover(:) = un_initialized
+      
+      if ( EDPftvarcon_inst%branch_turnover(ipft) > nearzero ) then
+         base_turnover(sapw_organ)   = hlm_freq_day / EDPftvarcon_inst%branch_turnover(ipft)
+         base_turnover(struct_organ) = hlm_freq_day / EDPftvarcon_inst%branch_turnover(ipft)
+         base_turnover(store_organ)  = hlm_freq_day / EDPftvarcon_inst%branch_turnover(ipft)
+      else
+         base_turnover(sapw_organ)   = 0.0_r8
+         base_turnover(struct_organ) = 0.0_r8
+         base_turnover(store_organ)  = 0.0_r8
+      end if
+
+      if ( EDPftvarcon_inst%root_long(ipft) > nearzero ) then
+         base_turnover(fnrt_organ) = hlm_freq_day / EDPftvarcon_inst%root_long(ipft)
+      else
+         base_turnover(fnrt_organ) = 0.0_r8
+      end if
+
+      if ( (EDPftvarcon_inst%leaf_long(ipft) > nearzero ) .and. &
+           (EDPftvarcon_inst%evergreen(ipft) == 1) ) then
+         base_turnover(leaf_organ) = hlm_freq_day / EDPftvarcon_inst%leaf_long(ipft)
+      else
+         base_turnover(leaf_organ) = 0.0_r8
+      endif
+
+      base_turnover(repro_organ)  = 0.0_r8
+
+      do i_var = 1, prt_instance%num_vars
+         
+         organ_id = prt_instance%state_descriptor(i_var)%organ_id
+         spec_id = prt_instance%state_descriptor(i_var)%spec_id
+
+         if ( any(spec_id == carbon_species) ) then
```

Comment:
what does the 'any' do here?

Reply:
The any will return true, if this particularly spec_id is equal to any of the ids in the carbon_species vector.